File: 49-GroupAnagrams/49-GroupAnagrams.py
# Last updated: 6/11/2026, 11:37:51 AM
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        big_dict = defaultdict(list)
        for i in strs:
            arr = [0]*26
            freqdict = Counter(i)
            for el in freqdict:
                arr[ord("a")-ord(el)] = freqdict[el]

            big_dict[tuple(arr)].append(i)

        return big_dict.values()


        

File: 49-GroupAnagrams/49-GroupAnagrams.cpp
// Last updated: 6/11/2026, 11:37:45 AM
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> ump;
        for(auto el: strs){
            string temp = el;
            sort(el.begin(), el.end());
            ump[el].push_back(temp);
        }

        vector<vector<string>> ans;

        for(auto el: ump){
            ans.push_back(el.second);
        }

        return ans;

        

    }
};